scipy/stats/stats.py

```diff
@@ -7324,13 +7324,18 @@ def kruskal(*args, nan_policy='propagate'):
 
     """
     args = list(map(np.asarray, args))
+
     num_groups = len(args)
     if num_groups < 2:
         raise ValueError("Need at least two groups in stats.kruskal()")
 
     for arg in args:
         if arg.size == 0:
             return KruskalResult(np.nan, np.nan)
+        elif np.squeeze(arg).ndim > 1:  # 0d (scalar) is ok
```

Comment:
in a few other tests (e.g. in _hypotests.py), an error is raised if `arg.ndim > 1`. I wonder if we should follow the same approach here, I did not understand the statement 'more than one non-singleton dimension' right away. it is a minor point, I am also fine with the current version.